s3fs/core.py

```diff
@@ -71,6 +71,59 @@ def split_path(path):
              'authenticated-read'}
 
 
+def title_case(string):
+    """
+    TitleCases a given string.
+
+    Parameters
+    ----------
+    string : underscore seperated string
+    """
+    return ''.join([x.capitalize() for x in string.split('_')])
+
+
+class SSEParams(object):
+
+    def __init__(self, server_side_encryption=None, sse_customer_algorithm=None,
+                 sse_customer_key=None, sse_kms_key_id=None):
+        self.ServerSideEncryption = server_side_encryption
+        self.SSECustomerAlgorithm = sse_customer_algorithm
+        self.SSECustomerKey = sse_customer_key
+        self.SSEKMSKeyId = sse_kms_key_id
+
+    def to_kwargs(self):
+        return {k: v for k, v in self.__dict__.items() if v is not None}
+
+
+class ParamKwargsHelper(object):
+    """
+    Utility class to help extract the subset of keys that an s3 method is
+    actually using
+
+    Parameters
+    ----------
+    s3 : S3FileSystem
+    """
+
+    def __init__(self, s3):
+        self.s3 = s3
+        self._kwarg_cache = {}
```

Comment:
Can different s3 instances have different values? I'm thinking not. Perhaps this should be a class-level dict, and the S3FileSystem instances won't need their own instances of ParamKwargsHelper, just call like a function.